### ISSUE TITLE
Mon interfaces not deleted after closing script normally

### ISSUE THREAD

**Issue by Edu4rdSHL** [v1s1t0r1sh3r3/airgeddon#141]
#### What is your airgeddon version?
8.01

#### What is your Linux O.S. and which version?

ArchLinux

#### Which is the chipset of your wireless card?

Ralink Technology, Corp. MT7601U Wireless Adapter

#### Describe the issue and the steps to reproduce it

When you put your Wireless Adapter in monitor mode, it create a new interface called **mon[1,2,3,4...]** and depending of the number of scans for targets it create a new one, the script works well, but when the script finished, the news interfaces are not deleted.

#### Screenshot

![2018-05-27-182220-sechacklabs](https://user-images.githubusercontent.com/32582878/40591685-388b781a-61db-11e8-8e72-9cc447cea28b.png)

#### Possible solution

Add to the script the line: `for newmoninterface in $( ls /sys/class/net/ |grep m ) ; do iw $newmoninterface del ; done`, it solved the issue.

**Comment by OscarAkaElvis** [comment]
Thanks for the report. For me is working ok I'll try to reproduce it under similar environment.... using Arch and not sure if could be related to your chipset. Is the problem happening with other wifi card? just to discard possibilities. Anyway I'll check to see if I can reproduce it in order to solve it.

**Comment by OscarAkaElvis** [comment]
Quite strange.... there is a line to clean up the devices on `restore_et_interface` function and usually it works very well... maybe on Arch is different. This line says:

`iw dev "${iface_monitor_et_deauth}" del > /dev/null 2>&1`

The unique difference between what you suggest is the "dev" parameter on the command. Can you confirm this command is not working for you? maybe you can remove the last part "> /dev/null 2>&1" to see if there are some errors.

I tested the command on a kali linux and it seems the same using the "dev" and launching it without the "dev" part... so it can be removed if is causing any problem on any linux.

**Comment by OscarAkaElvis** [comment]
Tested on Arch Linux, it's working fine. Not reproducible.

Anyway... airgeddon shouldn't create a "mon0" interface while setting the card into monitor mode. It's using airmon-zc if available and if not it uses airmon-ng. This makes that always the interface name changes, but it's not creating a new interface while setting the monitor mode. The unique place on which airgeddon is creating a new interface is on Evil Twin attacks and I tested the iw commands for creation and for deletion on Arch and everything is working fine.

Are you sure airgeddon is causing this?

**Comment by Edu4rdSHL** [comment]
Hi, @OscarAkaElvis. You can check it asciinema video: https://asciinema.org/a/e7MPRvtn1atjXYKleOqdUPWrL

**Comment by v1s1t0r1sh3r3** [comment]
We are not able to reproduce the problem. You say you are using Arch Linux but I see on your video that airgeddon is not able to detect it as an Arch Linux. It says "Unknown Linux". So I have some questions:

- What is the output of your `uname -a` command?
- Did you tried using other wifi card? same problem or is related to this wireless card?
- What is your aircrack suite version?

A last test you should do could be: Open airgeddon, put your card in monitor mode. Don't perform any other action, just exit airgeddon and launch this command to remove one of this interfaces: `iw dev "mon0" del` <- is the command working?

**Comment by Edu4rdSHL** [comment]
The script says "Unknown Linux" because my hardening process. I'm computer security and network privacy specialist, i'm sure that the script is reading /etc/issue for determine what distro i'm using, it's empty. The ` iw dev "$monIface" del` is working well, i modified the script and added my own line `for newmoninterface in $( ls /sys/class/net/ |grep mon ) ; do iw $newmoninterface del ; done` but it does not work either, i'm not sure if the kernel type have incidences there, i'm using linux-hardened but i'm able to reproduce it using linux-lts too and i think that is not related to kernel type. The aircrack version is the latest 1.2. Probably because the distro is unknown for the script it's skipping that order in https://github.com/v1s1t0r1sh3r3/airgeddon/blob/ae4a1e39635b35ee0ac7fcb25f19e99b71631b0b/airgeddon.sh#L1271

**Comment by v1s1t0r1sh3r3** [comment]
The `restore_et_function` is called only after finishing an evil twin attack. It's not called exiting the script because it's supposed that is not needed. The reason is because airgeddon detects airmon-ng and airmon-zc to change card mode. I'll explain:

On some systems, there is only present airmon-ng. On this systems there is no problem because using it, the interface name is changed but there is no new interface created while changing.

On other systems airmon-zc is present in addition of airmon-ng. On these systems there are two behaviors. airmon-zc is acting in the same way as airmon-ng is doing on systems with only airmon-ng. I mean, airmon-zc is changing the name fo the interface while changing its mode but it's not creating a new interface (as is in your case). That behavior (what is happening to you) is what airmon-ng is doing on systems with both (airmon-zc and airmon-ng).

Do you have airmon-zc on your system? if yes, it should be used instead of airmon-ng and it's supposed that airgeddon should select it automatically. If the answer is no... I can't understand why this is happening to you, sorry.

**Comment by Edu4rdSHL** [comment]
Thank you @v1s1t0r1sh3r3. My computer have airmon-ng, but dont know if it's symlinking to airmon-zc, in a couple of ours i'm giving the info to you (im not now in my computer).

**Comment by Edu4rdSHL** [comment]
Definitely aircrack is not the problem, I'm thinking it's a wifi adapter problem, but's weird because it's the only time that i'm having problems with that, i'm going to add for `newmoninterface in $( ls /sys/class/net/ |grep mon ) ; do iw $newmoninterface del ; done` at the end of the script for solve that problem. Thank you.

**Comment by OscarAkaElvis** [comment]
Ok, we can close the issue but it is a weird problem. Could be nice to know if using the same adapter is causing the same problem on other different system (a Debian or something like that).

Thanks for your report. Closing.
